scylla/src/cluster/worker.rs

```diff
@@ -467,6 +502,218 @@ impl ClusterWorker {
     fn update_cluster_state(&mut self, new_cluster_state: Arc<ClusterState>) {
         self.cluster_state.store(new_cluster_state);
     }
+
+    /// Handle node addition/removal/address changes.
+    ///
+    /// Emit respective events to the [HostListener], if configured.
+    fn handle_topology_changes(
+        known_peers: &HashMap<Uuid, Arc<Node>>,
+        new_known_peers: &HashMap<Uuid, Arc<Node>>,
+        #[cfg(all(scylla_unstable, feature = "unstable-host-listener"))] host_listener: Option<
+            &dyn HostListener,
+        >,
+        node_status: &mut HashMap<Uuid, NodeConnectivityStatus>,
+    ) {
```

Comment:
I'm not a fan of those `#[cfg(...)]` now being everywhere now. Idea: we can make HostListener exist unconditionally, but without the feature it will not be present in SessionConfig, and it will not be pub.
To make it non-pub, you can make the module containing it private, the items themselves pub, and add a pub re-export guarded by cfg.